Mathlib/Tactic/Congr!.lean

```diff
@@ -618,9 +618,20 @@ def Lean.MVarId.congrN! (mvarId : MVarId)
 where
   post (mvarId : MVarId) : CongrMetaM Unit := do
     for mvarId in ← mvarId.introsClean do
-      if let some mvarId ← mvarId.postCongr! config then
+      trace[congr!] "Attempting postCongr! {mvarId}"
+      let mvarId? ← try
+        mvarId.postCongr! config
+      catch e =>
+        -- I would prefer to just rethrow this error,
+        -- but it is somehow(!?) caught, and the mvarId is lost as a consequence.
+        -- throwTacticEx `congr! mvarId m!"internal error in postCongr!, {e.toMessageData}"
+        trace[congr!] "internal error in postCongr!, {e.toMessageData}"
+        throw e
+      match mvarId? with
+      | some mvarId =>
+        trace[congr!] "Recording {mvarId}."
         modify (fun s => {s with goals := s.goals.push mvarId})
-      else
+      | none =>
```

Comment:
Given Eric's fix to Convert.lean, do we need to make this change anymore? If we want logging, I think I'd rather that `postCongr!` itself have its body wrapped in a `try` block, where it re-throws the error but mentions that the error is passing through `postCongr!`.

Reply:
There's `Lean.Meta.throwNestedTacticEx` (check its implementation to see how simple it is -- if we want better context, then may as well just `throwError` like it does).

There's also the fancier `Lean.Elab.throwErrorWithNestedErrors`, which seems to actually report source positions of the nested errors. Mathlib never uses it directly, and instead it uses `Lean.Elab.nestedExceptionToMessageData` (see `focusAndDoneWithScope`)